### ISSUE TITLE
Display name for a content part field in a recipe is being ignored on import / export

### ISSUE THREAD

**Issue by rjpowers10** [OrchardCMS/OrchardCore#15062]
### Describe the bug
I'm trying to update to OC 1.8.1. In my recipe I specify a display name that is different than the technical name. This worked all the way up to OC 1.8, but in OC 1.8.1 my display name is being ignored and the technical name is used instead.

Similarly, if I set the desired display name through the UI and export the content part field definition into a recipe, the display name is set to the technical name.

### To Reproduce
1. Create a new OC 1.8.1 site using the blog recipe
2. Create a new content type as follows: technical name "MyContentType", display name "My Content Type".
3. Add a boolean field to the content type as follows: technical name "MyBooleanField", display name "My Boolean Field"
4. Go to /Admin/DeploymentPlan/Index and create a new deployment plan. Add the "Update Content Definitions" step and check the box for "My Content Type"
5. Execute the deployment plan

### Expected behavior
The resulting recipe has the correct technical and display names for the content type, but not the content part.
```
{
  "name": "",
  "displayName": "",
  "description": "",
  "author": "",
  "website": "",
  "version": "",
  "issetuprecipe": false,
  "categories": [],
  "tags": [],
  "steps": [
    {
      "name": "ContentDefinition",
      "ContentTypes": [
        {
          "Name": "MyContentType",
          "DisplayName": "My Content Type",
          "Settings": {
            "ContentTypeSettings": {
              "Creatable": true,
              "Listable": true,
              "Draftable": true,
              "Versionable": true,
              "Securable": true
            },
            "FullTextAspectSettings": {}
          },
          "ContentTypePartDefinitionRecords": [
            {
              "PartName": "MyContentType",
              "Name": "MyContentType",
              "Settings": {
                "ContentTypePartSettings": {
                  "Position": "0"
                }
              }
            }
          ]
        }
      ],
      "ContentParts": [
        {
          "Name": "MyContentType",
          "Settings": {},
          "ContentPartFieldDefinitionRecords": [
            {
              "FieldName": "BooleanField",
              "Name": "MyBooleanField",
              "Settings": {
                "ContentPartFieldSettings": {
                  "DisplayName": "MyBooleanField",
                  "Position": "0"
                }
              }
            }
          ]
        }
      ]
    }
  ]
}
```

### To Reproduce the other issue
Now, I start a brand new site from scratch. Instead of creating the content type from the UI, I import it from the following recipe (note the corrected display name). Go to /Admin/DeploymentPlan/Import/Json and paste in this recipe.
```
{
  "name": "",
  "displayName": "",
  "description": "",
  "author": "",
  "website": "",
  "version": "",
  "issetuprecipe": false,
  "categories": [],
  "tags": [],
  "steps": [
    {
      "name": "ContentDefinition",
      "ContentTypes": [
        {
          "Name": "MyContentType",
          "DisplayName": "My Content Type",
          "Settings": {
            "ContentTypeSettings": {
              "Creatable": true,
              "Listable": true,
              "Draftable": true,
              "Versionable": true,
              "Securable": true
            },
            "FullTextAspectSettings": {}
          },
          "ContentTypePartDefinitionRecords": [
            {
              "PartName": "MyContentType",
              "Name": "MyContentType",
              "Settings": {
                "ContentTypePartSettings": {
                  "Position": "0"
                }
              }
            }
          ]
        }
      ],
      "ContentParts": [
        {
          "Name": "MyContentType",
          "Settings": {},
          "ContentPartFieldDefinitionRecords": [
            {
              "FieldName": "BooleanField",
              "Name": "MyBooleanField",
              "Settings": {
                "ContentPartFieldSettings": {
                  "DisplayName": "My Boolean Field",
                  "Position": "0"
                }
              }
            }
          ]
        }
      ]
    }
  ]
}
```
Go to /Admin/ContentParts/MyContentType/Fields/MyBooleanField/Edit and note the display name is incorrectly set to the technical name.

![image](https://github.com/OrchardCMS/OrchardCore/assets/68876423/fd94ff79-c57c-475b-9887-e2a339f7a782)

@MikeAlhayek could this be related to #15013?

**Comment by MikeAlhayek** [comment]
@rjpowers10 this issue was in 1.8.0. It was fixed in 1.8.1. Are you able to test 1.8.1 and see if that solves the problem?

Related to #14985 .

**Comment by rjpowers10** [comment]
@MikeAlhayek I was in the process of updating to OC 1.8, and I did not encounter the issue then. It was only after I did another update to 1.8.1 that I got the issue. After I noticed the issue in my app I tried to reproduce it in a plan old OC 1.8.1 app (as described in my initial post) and I got the same issue there.
